src/alembic_utils/replaceable_entity.py

```diff
@@ -200,167 +281,199 @@ def register_entities(
-            if len(function_group.collect()) > 1:
-                raise DuplicateRegistration(
-                    f"PGFunction with identity {ident} was registered multiple times"
-                )
+@comparators.dispatch_for("schema")
+def compare_registered_entities(
+    autogen_context: AutogenContext,
+    upgrade_ops,
+    schemas: List[Optional[str]],
+):
+    connection = autogen_context.connection
 
-        # https://alembic.sqlalchemy.org/en/latest/autogenerate.html#controlling-what-to-be-autogenerated
-        # if EnvironmentContext.configure.include_schemas is True, all non-default "scehmas" should be included
-        # pulled from the inspector
-        include_schemas: bool = autogen_context.opts["include_schemas"]
+    entities = registry.entities()
 
-        reflected_schemas = set(
-            autogen_context.inspector.get_schema_names() if include_schemas else []  # type: ignore
-        )
-        sqla_schemas: Set[Optional[str]] = sqla_schemas or {}  # type: ignore
-        manual_schemas = set(schemas or [])  # Deprecated for remove in 0.6.0
-        entity_schemas = {x.schema for x in entities}  # from ReplaceableEntity instances
-        all_schema_references = reflected_schemas | sqla_schemas | manual_schemas | entity_schemas  # type: ignore
+    # https://alembic.sqlalchemy.org/en/latest/autogenerate.html#controlling-what-to-be-autogenerated
+    # if EnvironmentContext.configure.include_schemas is True, all non-default "schemas" should be included
+    # pulled from the inspector
+    include_schemas: bool = autogen_context.opts["include_schemas"]
 
-        # Remove excluded schemas
-        observed_schemas: Set[str] = {
+    reflected_schemas = set(
+        autogen_context.inspector.get_schema_names() if include_schemas else []  # type: ignore
+    )
+    sqla_schemas: Set[Optional[str]] = set(schemas)
+    manual_schemas = set(registry.schemas or set())  # Deprecated for remove in 0.6.0
+    entity_schemas = {x.schema for x in entities}  # from ReplaceableEntity instances
+    all_schema_references = reflected_schemas | sqla_schemas | manual_schemas | entity_schemas  # type: ignore
+
+    # Remove excluded schemas
+    observed_schemas: Set[str] = {
+        schema_name
+        for schema_name in all_schema_references
+        if (
             schema_name
-            for schema_name in all_schema_references
-            if (
-                schema_name
-                not in (exclude_schemas or [])  # user defined. Deprecated for remove in 0.6.0
-                and schema_name not in {"information_schema", None}
-            )
-        }
+            is not None
+            not in (
+                registry.exclude_schemas or set()
+            )  # user defined. Deprecated for remove in 0.6.0
+            and schema_name not in {"information_schema", None}
+        )
+    }
+
+    if any(e.depends_on for e in entities):
```

Comment:
This looks like if a user were to use `depends_on` in any of their entities, all dependencies are assumed to be explicit